Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java

```diff
@@ -17,79 +21,205 @@
+            if (node instanceof PredicateNode) {
+                // the key does not matter
+                // as it is never being used
+                updatedNodes.put(String.valueOf(++index), node);
+                continue;
+            }
+
+            // node is subject or object
+            // If the concept id is already present in the
+            // updated nodes map - dont add it
+            if (!updatedNodes.containsKey(node.getConceptId())) {
+                updatedNodes.put(node.getConceptId(), node);
+            }
+        }
+
+        // update the links to point to the new updatedNodes
+        for (Link link: links) {
+            Node subjectNode = link.getSubject();
+            if (!(subjectNode instanceof PredicateNode)) {
+                link.setSubject(updatedNodes.get(subjectNode.getConceptId()));
+            }
+
+            Node objectNode = link.getObject();
+            if (!(objectNode instanceof PredicateNode)) {
+                link.setObject(updatedNodes.get(objectNode.getConceptId()));
+            }
+        }
+
+        // return new network with updated nodes and links
+        return new TransformedNetwork(updatedNodes, links);
+    }
+
+    @Override
+    public ITransformedNetwork getSearchTransformedNetwork(String projectId, String conceptId)
+        throws QuadrigaStorageException {
+        // get the transformed network and search for the concept id
+        List<INetwork> networkList = getNetworkList(projectId);
+
+        if (networkList == null) {
+            return null;
+        }
+
+        ITransformedNetwork transformedNetwork = getTransformedNetworkOfProject(projectId);
+        // separate each statement id and include all the list of nodes in the statement id
+        Map<String, Set<Node>> statementIdMap = getStatementIdNodeMap(transformedNetwork.getNodes());
+
+        // Filter the nodes with the concept id
+        List<Node> searchNodes = new ArrayList<Node>();
+        for (Node node: transformedNetwork.getNodes().values()) {
+            if (conceptId.equals(node.getConceptId())) {
+                searchNodes.add(node);
+            }
+        }
+
+        // include only those statements which have the search node
+        Iterator<Map.Entry<String, Set<Node>>> it = statementIdMap.entrySet().iterator();
+        Set<Node> searchedNodeSet = new HashSet<Node>();
+        while (it.hasNext()) {
+            boolean isNodeExists = false;
+            Set<Node> nodeSet = it.next().getValue();
+            // search if the node exists
+            for (Node node: searchNodes) {
+                if (nodeSet.contains(node)) {
+                    // include this statement
+                    isNodeExists = true;
+                }
+            }
+            // remove the statement if it is not present
+            if (!isNodeExists) {
+                it.remove();
+            } else {
+                searchedNodeSet.addAll(nodeSet);
+            }
+        }
```

Comment:
It makes sense. This is the final change I made.
1. I search for all the nodes which I whose concept id matches the search concept id and then store all the statement ids in a hash set.
2. I will iterate over all the nodes and check if there is a statement that matches the search statement id set.
3. If it matches I will add that to a final node map.
4. I added a statement id field in Link class.
5. I will iterate through all the links and check if the statement id exists in the statement id set.